### PR TITLE
Implemented New API + Follow along algorithm

### PR DESCRIPTION
**Pull request by fdalvi**
- Updated API to follow new spec
- Updated test suite to follow new spec
- Implemented new ayah search + follow along algorithm

### CODE REVIEW

**Comment by fdalvi** [comment]
Including Anas in the PR incase he sees something relevant to the Frontend work being done.

Kind of rushed this one out - so might have missed something in terms of code style/consistency - so please feel free to comment all nitty gritty

File: src/transcribe.js [hunk 1]
```diff
@@ -1,34 +1,327 @@
...
+            // End of ayah - start looking for next ayah
+            if (process.env.DEBUG === 'development') {
+                console.log("[Follow along] Detected end of ayah")
+            }
+            this.onMatchFound(this.currentSurahNum, this.currentAyahNum, matchedWords.length);
+            this.currentPartialAyahIndex = Math.min(
+                this.currentPartialAyahIndex + transcript.length + finalSlack + 1,
+                this.partialTranscript.length
+            );
+            // console.log(this.partialTranscript.length + " " + this.currentPartialAyahIndex);
```
Comment (Ananasship) [inline src/transcribe.js]: nit: dead code

File: src/transcribe.js [hunk 2]
```diff
@@ -1,34 +1,327 @@
...
+                console.log(err);
+            }
+            return callback([]);
+        });
+    };
+
+    /**
+     * Quran text retrieval function for a single ayah
+     *
+     * ayah: object
```
Comment (Ananasship) [inline src/transcribe.js]: nit: spacing

File: src/transcribe.js [hunk 3]
```diff
@@ -1,34 +1,327 @@
...
+     *  suranNum: int
+     *  ayahNum: int
+     *
+     * callback: function
+     *  signature: (error, results)
+     */
+    getAyah = (ayah, callback) => {
+        fetch(`${api_constants.TARTEEL_API}/?surah=${ayah.surahNum}&ayah=${ayah.ayahNum}`, {
```
Comment (Ananasship) [inline src/transcribe.js]: +1 for refactoring these out in the future

File: test/suites/multiAyatTests.js [hunk 4]
```diff
@@ -2,15 +2,17 @@ import { expect } from 'chai';
...
 // Import utilities
-import { streamMultipleAudioInRealtime } from '../utils';
+import { streamMultipleAudioInRealtime, stopAllStreaming } from '../utils';
 import { loadAudioFiles } from '../utils';
 
 export default function suite (mochaContext, socketUrl, options) {
   mochaContext.timeout(50000);
-  let client1,
-      ayatData = [null, null, null],
-      ayatText = ['بسم الله الرحمن الرحيم','الحمد لله رب العالمين','الرحمن الرحيم'],
```
Comment (Ananasship) [inline test/suites/multiAyatTests.js]: nice

**Pull request merged**